cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/cost/AwsPricingCache.java

```diff
@@ -0,0 +1,85 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.cost;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.amazonaws.services.pricing.AWSPricing;
+import com.amazonaws.services.pricing.AWSPricingClientBuilder;
+import com.amazonaws.services.pricing.model.Filter;
+import com.amazonaws.services.pricing.model.FilterType;
+import com.amazonaws.services.pricing.model.GetProductsRequest;
+import com.amazonaws.services.pricing.model.GetProductsResult;
+import com.amazonaws.services.pricing.model.NotFoundException;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.collect.HashBasedTable;
+import com.google.common.collect.Table;
+import com.sequenceiq.cloudbreak.cloud.aws.common.cost.model.PriceListElement;
+
+@Service("awsPricingCache")
+public class AwsPricingCache {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsPricingCache.class);
+
+    private static final String PRICING_API_ENDPOINT_REGION = "us-east-1";
+
+    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+    private final AWSPricing awsPricing = AWSPricingClientBuilder.standard().withRegion(PRICING_API_ENDPOINT_REGION).build();
+
+    private final Table<String, String, PriceListElement> cache = HashBasedTable.create();
+
+    public double getPriceForInstanceType(String region, String instanceType) {
+        PriceListElement priceListElement = getPriceList(region, instanceType);
+        return priceListElement.getTerms().getOnDemand().values().stream().findFirst().orElseThrow(() ->
+                        new NotFoundException(String.format("Couldn't find the price for region [%s] and instance type [%s].", region, instanceType)))
+                .getPriceDimensions().values().stream().findFirst().orElseThrow(() ->
+                        new NotFoundException(String.format("Couldn't find the price for region [%s] and instance type [%s].", region, instanceType)))
+                .getPricePerUnit().getUsd();
+    }
+
+    public int getCpuCountForInstanceType(String region, String instanceType) {
+        PriceListElement priceListElement = getPriceList(region, instanceType);
+        return priceListElement.getProduct().getAttributes().getVcpu();
+    }
+
+    public int getMemoryForInstanceType(String region, String instanceType) {
+        PriceListElement priceListElement = getPriceList(region, instanceType);
+        String memory = priceListElement.getProduct().getAttributes().getMemory();
+        return Integer.parseInt(memory.replaceAll("\\D", ""));
+    }
+
+    private PriceListElement getPriceList(String region, String instanceType) {
```

Comment:
maybe we should caching this

Reply:
The `getPriceList` method checks, if we already have a PriceListElement object in the cache, for the specified _region - instanceType_ combination.
The price, cpu and memory information are all in the PriceListElement object.